services/state/roster/roster.proto

```diff
@@ -0,0 +1,110 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "basic_types.proto";
+
+option java_package = "com.hederahashgraph.api.proto.java";
+// <<<pbj.java_package = "com.hedera.hapi.node.state.roster">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * A single roster in the network state.
+ * <p>
+ * The roster SHALL be a list of `RosterEntry` objects.
+ */
+message Roster {
+
+    /**
+     * List of roster entries, one per consensus node.
+     * <p>
+     * This list SHALL contain roster entries in natural order of ascending node ids.
+     * This list SHALL NOT be empty.<br/>
+     */
+    repeated RosterEntry rosters = 1;
+}
+
+/**
+ * A single roster entry in the network state.
+ *
+ * Each roster entry SHALL encapsulate the elements required
+ * to manage node participation in the Threshold Signature Scheme (TSS).<br/>
+ * All fields except tss_encryption_key are REQUIRED.
+ */
+message RosterEntry {
+
+    /**
+     * A consensus node identifier.
+     * <p>
+     * Node identifiers SHALL be unique _within_ a ledger,
+     * and MAY NOT be repeated _between_ shards and realms.
+     */
+    uint64 node_id = 1;
+
+    /**
+     * A consensus weight.
+     * <p>
+     * Each node SHALL have a weight of zero or more in consensus calculations.<br/>
+     * The sum of the weights of all nodes in the roster SHALL form the total weight of the system,<br/>
+     * and each node's individual weight SHALL be proportional to that sum.<br/>
+     */
+    uint64 weight = 2;
+
+    /**
+     * An RSA public certificate used for signing gossip events.
+     * <p>
+     * This value SHALL be a certificate of a type permitted for gossip
+     * signatures.<br/>
+     * This value SHALL be the DER encoding of the certificate presented.<br/>
+     * This field is REQUIRED and MUST NOT be empty.
+     */
+    bytes gossip_ca_certificate = 3;
+
+    /**
+     * An ALT_BN128 elliptic curve public encryption key.<br/>
+     * <p>
+     * The elliptic curve type may change in the future. For example, <br/>
+     * if the Ethereum ecosystem creates precompiles for BLS12_381,
+     * we may switch to that curve.
+     * <p>
+     * This value SHALL be specified according to EIP-196 and EIP-197 standards, <br/>
+     * See https://eips.ethereum.org/EIPS/eip-196#encoding and <br/>
+     * https://eips.ethereum.org/EIPS/eip-197#encoding
+     * <p>
+     * This field is OPTIONAL (that is, it can initially be null)
+     * but once set, it can no longer be null.
```

Comment:
For _fields_ we should use line breaks between individual requirements, and a single paragraph break between the summary/description and the requirements.
```suggestion
     * This value SHALL be specified according to EIP-196 and EIP-197 standards,
     * See https://eips.ethereum.org/EIPS/eip-196#encoding and
     * https://eips.ethereum.org/EIPS/eip-197#encoding<br/>
     * This field is _initially_ OPTIONAL (i.e. it can be unset _when created_)
     * but once set, it is REQUIRED thereafter.
```